action.yml

```diff
@@ -25,21 +25,42 @@ inputs:
 runs:
   using: "composite"
   steps:
-    - uses: actions/setup-java@v4
+    - name: Set variables
+      id: variables
+      run: |
+        OPENFIRE_VERSION=${{ inputs.version }}
+        echo "OPENFIRE_VERSION=${OPENFIRE_VERSION}" >> $GITHUB_OUTPUT
+        echo "OPENFIRE_VERSION_UNDERSCORE=${OPENFIRE_VERSION//./_}" >> $GITHUB_OUTPUT
+
+        if [[ "${{ inputs.daily }}" == "false" && ( "${{ inputs.version }}" == 3.* || "${{ inputs.version }}" == 4.* ) ]]; then
+          echo "JAVA_VERSION=11" >> $GITHUB_OUTPUT
+        else
+          echo "JAVA_VERSION=17" >> $GITHUB_OUTPUT
+        fi
+      shell: bash
+
+    - name: Setup Java 11
+      uses: actions/setup-java@v4
+      if: ${{ steps.variables.outputs.JAVA_VERSION == '11' }}
       with:
         java-version: 11
```

Comment:
Can we replace `java-version: 11` with something like `java-version: ${{ steps.variables.outputs.JAVA_VERSION }}` and remove the duplicated blocks for java 17?

Reply:
Oooh. Of course! Good idea!

Reply:
Awesome!